install.sh

```diff
@@ -153,6 +153,9 @@ setup_env() {
                 if [ -z "${K3S_TOKEN}" ] && [ -z "${K3S_CLUSTER_SECRET}" ]; then
                     fatal "Defaulted k3s exec command to 'agent' because K3S_URL is defined, but K3S_TOKEN or K3S_CLUSTER_SECRET is not defined."
                 fi
+                if ! echo "${K3S_URL}" | grep -q "^https://"; then
+                    fatal "Only https:// URLs are supported for K3S_URL (have ${K3S_URL})"
+                fi
```

Comment:
To me it would feel more natural to do this with simple shell built-in case functionality rather than involving echo and grep on the outside.

```
case "${K3S_URL}" in
    https://*)
        ;;
    *)
        fatal "Only https:// URLs are supported for K3S_URL (have ${K3S_URL})"
    ;;
esac
```

Reply:
Not quite as concise, but I dig it. Always a fan of using builtins. Thanks!